internal/services/access_targets.go

```diff
@@ -0,0 +1,194 @@
+package services
+
+import (
+	"context"
+	"fmt"
+	"github.com/apono-io/apono-sdk-go"
+	"github.com/apono-io/terraform-provider-apono/internal/models"
+	"github.com/apono-io/terraform-provider-apono/internal/utils"
+	"github.com/hashicorp/terraform-plugin-framework/diag"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	"golang.org/x/exp/slices"
+)
+
+func ConvertIntegrationTargetsApiToTerraformModel(ctx context.Context, aponoClient *apono.APIClient, integrationTargets []apono.AccessTargetIntegrationV1) ([]models.IntegrationTarget, diag.Diagnostics) {
+	availableIntegrations, _, err := aponoClient.IntegrationsApi.ListIntegrationsV2(ctx).Execute()
+	if err != nil {
+		return nil, utils.GetDiagnosticsForApiError(err, "list", "integrations", "")
+	}
+
+	var dataIntegrationTargets []models.IntegrationTarget
+	for _, integrationTarget := range integrationTargets {
+		integration, diagnostics := convertIntegrationTargetApiToTerraformModel(ctx, &integrationTarget, availableIntegrations.Data)
+		if len(diagnostics) > 0 {
+			return nil, diagnostics
+		}
+
+		dataIntegrationTargets = append(dataIntegrationTargets, *integration)
+	}
+
+	return dataIntegrationTargets, nil
+}
+
+func ConvertIntegrationTargetsTerraformModelToApi(ctx context.Context, aponoClient *apono.APIClient, integrationTargets []models.IntegrationTarget) ([]apono.AccessTargetIntegrationV1, diag.Diagnostics) {
+	availableIntegrations, _, err := aponoClient.IntegrationsApi.ListIntegrationsV2(ctx).Execute()
+	if err != nil {
+		return nil, utils.GetDiagnosticsForApiError(err, "list", "integrations", "")
+	}
+
+	var resultIntegrationTargets []apono.AccessTargetIntegrationV1
+	for _, integrationTarget := range integrationTargets {
+		integration, diagnostics := convertIntegrationTargetTerraformModelToApi(integrationTarget, availableIntegrations.Data)
+		if len(diagnostics) > 0 {
+			return nil, diagnostics
+		}
+
+		resultIntegrationTargets = append(resultIntegrationTargets, *integration)
+	}
+
+	return resultIntegrationTargets, nil
+}
```

Comment:
moved logic from `services/access_flow.go`

Reply:
lets be doing such moves in separate PRs next time